File: lecture1/problems/mario/mario.c
#include <cs50.h>
#include <stdio.h>

int main()
{
    for (int i = 0; i < 8; i++) {
        for (int j = 7; j >= 0; j--) {
            if (j <= i) {
                printf("#");
            } else {
                printf(" ");
            }
        }
        printf("\n");
    }
}

File: lecture6/problems/sentimental-mario-more/mario.py
"""Pyramid's print 
Print the pyramid according to heigh passed by user
"""

from cs50 import get_int


def main() -> None:
    """Ask height for the user and print using print_pyramid()"""
    while True:
        height = get_int('Height:')

        if height is None:
            continue
        elif height < 1 or height > 8:
            continue
        break

    print_pyramid(height)


def print_pyramid(height: int) -> None:
    """Print a pyramide with height passed by argument"""
    for i in range(1, height + 1):
        for j in reversed(range(1, height + 1)):
            if j <= i:
                print('#', end='')
            else:
                print(' ', end='')
        print('  ', end='')

        for j in range(i):
            print('#', end='')
        print()


if __name__ == '__main__':
    main()
